src/architecture/architect.js

```diff
@@ -5,562 +5,596 @@ var Layer = require('./layer');
-  */
-  Perceptron: function () {
-    // Convert arguments to Array
-    var layers = Array.prototype.slice.call(arguments);
-    if (layers.length < 3) {
-      throw new Error('You have to specify at least 3 layers');
+    for (j = 0; j < nodes[i].connections.gated.length; j++) {
+      network.gates.push(nodes[i].connections.gated[j]);
+    }
+    if (nodes[i].connections.self.weight !== 0) {
+      network.selfconns.push(nodes[i].connections.self);
     }
+  }
 
-    // Create a list of nodes/groups
-    var nodes = [];
-    nodes.push(new Group(layers[0]));
+  network.nodes = nodes;
 
-    for (var i = 1; i < layers.length; i++) {
-      var layer = layers[i];
-      layer = new Group(layer);
-      nodes.push(layer);
-      nodes[i - 1].connect(nodes[i], methods.connection.ALL_TO_ALL);
-    }
+  return network;
+},
 
-    // Construct the network
-    return architect.Construct(nodes);
-  },
-
-  /**
-  * Creates a randomly connected network
-  *
-  * @param {number} input Number of input nodes
-  * @param {number} hidden Number of nodes inbetween input and output
-  * @param {number} output Number of output nodes
-  * @param {object} [options] Configuration options
-  * @param {number} [options.connections=hidden*2] Number of connections (Larger than hidden)
-  * @param {number} [options.backconnections=0] Number of recurrent connections
-  * @param {number} [options.selfconnections=0] Number of self connections
-  * @param {number} [options.gates=0] Number of gates
-  *
-  * @example
-  * let { architect } = require("@liquid-carrot/carrot");
-  *
-  * let network = architect.Random(1, 20, 2, {
-  *   connections: 40,
-  *   gates: 4,
-  *   selfconnections: 4
-  * });
-  *
-  * @returns {Network}
-  */
-  Random: function (input, hidden, output, options) {
-    options = options || {};
-
-    var connections = options.connections || hidden * 2;
-    var backconnections = options.backconnections || 0;
-    var selfconnections = options.selfconnections || 0;
-    var gates = options.gates || 0;
-
-    var network = new Network(input, output);
-
-    var i;
-    for (i = 0; i < hidden; i++) {
-      network.mutate(methods.mutation.ADD_NODE);
-    }
+/**
+* Creates a multilayer perceptron (MLP)
+*
+* @param {...number} layerNeurons Number of neurons in input layer, hidden layer(s), and output layer as a series of numbers (min 3 arguments)
+*
+* @example
+* let { architect } = require("@liquid-carrot/carrot");
+*
+* // Input 2 neurons, Hidden layer: 3 neurons, Output: 1 neuron
+* let myPerceptron = new architect.Perceptron(2,3,1);
+*
+* // Input: 2 neurons, 4 Hidden layers: 10 neurons, Output: 1 neuron
+* let myPerceptron = new architect.Perceptron(2, 10, 10, 10, 10, 1);
+*
+* @returns {Network} Feed forward neural network
+*/
+Perceptron: function () {
+  // Convert arguments to Array
+  // Coerce arguments to be Array-like w/ .call
+  var sizes = Array.prototype.slice.call(arguments); //type-casting/coercion arguments into array
```

Comment:
You can use the newer `Array.from` instead these days.

Reply:
https://stackoverflow.com/questions/36995617/convert-array-like-objects-array-prototype-slice-or-array-from